### PR TITLE
CORE-5102: Flow marker interface and generic marshalling service

### PR DESCRIPTION
**Pull request by JamesHR3**
This PR is a follow up to the previous flow API changes that introduces a common `Flow` marker interface and a generic marshalling service that the `JsonMarshallingService` inherits from. Some further tidying up is also present.

### CODE REVIEW

**Comment by JamesHR3** [comment]
Draft while I double check what other changes I need to make.

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/sandbox/SandboxDependencyInjector.kt [hunk 1]
```diff
@@ -13,7 +14,7 @@ interface SandboxDependencyInjector : AutoCloseable, NonSerializable {
      * Set any property on the flow marked with @[CordaInject] with an instance of the type specified.
      * @param flow The flow to receive the injected services.
      */
-    fun injectServices(flow: Any)
+    fun injectServices(flow: Flow)
```
Comment (rick-r3) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/sandbox/SandboxDependencyInjector.kt:17]: Should this be flow specific?  i.e. can we just inject into anything?

As the comments / naming are written, this is correct as is.

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/sandbox/impl/SandboxDependencyInjectorImpl.kt [hunk 2]
```diff
@@ -32,7 +33,7 @@ class SandboxDependencyInjectorImpl(
         closeable.close()
     }
 
-    override fun injectServices(flow: Any) {
+    override fun injectServices(flow: Flow) {
```
Comment (rick-r3) [inline components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/sandbox/impl/SandboxDependencyInjectorImpl.kt:36]: As per interface

File: components/flow/flow-service/src/integrationTest/kotlin/net/corda/flow/testing/fakes/FakeSandboxGroupContextComponent.kt [hunk 3]
```diff
@@ -103,7 +104,7 @@ class FakeSandboxGroupContextComponent : SandboxGroupContextComponent {
     }
 
     class FakeSandboxDependencyInjector : SandboxDependencyInjector {
-        override fun injectServices(flow: Any) {
+        override fun injectServices(flow: Flow) {
```
Comment (rick-r3) [inline components/flow/flow-service/src/integrationTest/kotlin/net/corda/flow/testing/fakes/FakeSandboxGroupContextComponent.kt:107]: See comment about interface.

File: components/flow/flow-service/src/test/kotlin/net/corda/flow/fiber/RPCRequestDataImplTest.kt [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Test
+    fun `rpc request data is retrieved when getRequestBody is called`() {
+        val requestData = RPCRequestDataImpl(setupStartArgs())
+        assertEquals(TEST_START_ARGS, requestData.getRequestBody())
+    }
+
+    @Test
+    fun `rpc request data can marshal request body to a type`() {
```
Comment (rick-r3) [inline components/flow/flow-service/src/test/kotlin/net/corda/flow/fiber/RPCRequestDataImplTest.kt:30]: Add test for List method?

Comment (JamesHR3) [reply components/flow/flow-service/src/test/kotlin/net/corda/flow/fiber/RPCRequestDataImplTest.kt:30]: ...which I forgot to add to the API PR. :facepalm: I'll follow up.

Comment (JamesHR3) [reply components/flow/flow-service/src/test/kotlin/net/corda/flow/fiber/RPCRequestDataImplTest.kt:30]: One advantage of backing the API PR out is that I've now addressed this.

File: components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/sandbox/SandboxDependencyInjector.kt (hunk 1 above)
Comment (JamesHR3) [reply components/flow/flow-service/src/main/kotlin/net/corda/flow/pipeline/sandbox/SandboxDependencyInjector.kt:17]: I'd like to leave it as `Flow` for now until we know what else we're injecting into. Making it `Any` would imply we also need to move it out of the flow engine implementation I think.

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**